server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/CassandraJmapExtension.java

```diff
@@ -43,19 +46,71 @@
 import org.junit.jupiter.api.extension.ParameterResolver;
 import org.junit.rules.TemporaryFolder;
 
+import com.github.fge.lambdas.Throwing;
+
 public class CassandraJmapExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback, ParameterResolver {
+    public interface JamesLifeCyclePolicy {
+        JamesLifeCyclePolicy EACH = serverSupplier -> new JamesLifecycleHandler(
+            Optional::empty,
+            () -> Optional.of(serverSupplier.get()),
+            GuiceJamesServer::stop,
+            guiceJamesServer -> { });
+        JamesLifeCyclePolicy ALL = serverSupplier -> new JamesLifecycleHandler(
+            () -> Optional.of(serverSupplier.get()),
+            Optional::empty,
+            guiceJamesServer -> { },
+            GuiceJamesServer::stop);
```

Comment:
the naming is a little hard to get out of context. s/EACH/FOR_EACH_TEST/ s/ALL/COMMON_TO_ALL_TEST/ ? Or something like that.

Reply:
What about having a builder for JamesLifecycleHandler, the parameters are really not straightforward.